tools/build_defs/repo/git_worker.bzl

```diff
@@ -200,11 +200,22 @@ def _execute(ctx, git_repo, args):
     # See https://github.com/bazelbuild/bazel/issues/21438
     start = ["git", "-c", "core.fsmonitor=false"]
     return ctx.execute(
-        start + args,
+        _strip_git_vars(start + args),
         environment = ctx.os.environ,
         working_directory = str(git_repo.directory),
     )
 
+def _strip_git_vars(args):
+    """ Wraps args to strip out all Git local environment variables.
+
+    This ensures that these variables have no effect on Git's operation.
+    """
+    return [
+        "bash",
+        "-c",
+        "unset $(git rev-parse --local-env-vars); " + " ".join(args),
```

Comment:
Is it clear that unsetting all `GIT_` vars is correct? What about something like `GIT_SSH` or `GIT_SSL_NO_VERIFY`?

Reply:
```shell
$ git rev-parse --local-env-vars
GIT_ALTERNATE_OBJECT_DIRECTORIES
GIT_CONFIG
GIT_CONFIG_PARAMETERS
GIT_CONFIG_COUNT
GIT_OBJECT_DIRECTORY
GIT_DIR
GIT_WORK_TREE
GIT_IMPLICIT_WORK_TREE
GIT_GRAFT_FILE
GIT_INDEX_FILE
GIT_NO_REPLACE_OBJECTS
GIT_REPLACE_REF_BASE
GIT_PREFIX
GIT_INTERNAL_SUPER_PREFIX
GIT_SHALLOW_FILE
GIT_COMMON_DIR
```
So this doesn't snarf `GIT_SSH` or `GIT_SSL_NO_VERIFY`. I'll dig info the Windows issue.